src/RazorSlices/RazorSliceHttpResultOfTModel.cs

```diff
@@ -38,24 +39,27 @@ Task IResult.ExecuteAsync(HttpContext httpContext)
     {
         ArgumentNullException.ThrowIfNull(httpContext);
 
+        var htmlEncoder = HtmlEncoder ?? httpContext.RequestServices.GetService<HtmlEncoder>();
+
         httpContext.Response.StatusCode = StatusCode;
         httpContext.Response.ContentType = ContentType;
         httpContext.Response.RegisterForDispose(this);
 
-        var renderTask = this.RenderToPipeWriterAsync(httpContext.Response.BodyWriter, HtmlEncoder);
+#pragma warning disable CA2012 // Use ValueTasks correctly: The ValueTask is observed in code below
+        var renderTask = this.RenderToPipeWriterAsync(httpContext.Response.BodyWriter, htmlEncoder, httpContext.RequestAborted);
+#pragma warning restore CA2012
 
-        if (renderTask.IsCompletedSuccessfully)
+        if (renderTask.HandleSynchronousCompletion())
         {
-            renderTask.GetAwaiter().GetResult();
-            return httpContext.Response.BodyWriter.FlushAsync().AsTask();
+            return httpContext.Response.BodyWriter.FlushAsync(httpContext.RequestAborted).AsTask();
```

Comment:
I _think_ this can allocate since it'll create a `Task<FlushResult>` even though we're throwing away the result. See https://github.com/dotnet/aspnetcore/blob/main/src/Shared/ValueTaskExtensions/ValueTaskExtensions.cs for what we do in ASP.NET Core when ignoring the result to avoid the allocation.